File: C Projects/Introduction to programming in C language/pointers/calculator.c
#include<stdio.h>

int add(int a, int b);
int subtract(int a, int b);
int multiply(int a, int b);
int divide(int a, int b);

void main(void){

    int num1, num2, op;
    int (*ptr[4])(int, int) ={&add, &subtract, &multiply, divide};

    printf("Enter first number: ");
    scanf("%d", &num1);
    printf("Enter second number: ");
    scanf("%d", &num2);
    printf("Please select operation (1-add, 2-subtract, 3-multiply, 4-divide): ");
    scanf("%d", &op);

    int re = ptr[op-1](num1,num2);

    printf("%d %c %d = %d.\n",num1,((op==1)?'+':(op==2)?'-':(op==3)?'*':'/'),num2,re);


}
int add(int a, int b){
    return(a+b);
}
int subtract(int a, int b){
    return(a-b);
}
int multiply(int a, int b){
    return(a*b);
}
int divide(int a, int b){
    return(a/b);
}

File: Java Projects/Java 102 course/src/examples/calculator/Calculator.java
package examples.calculator;

public class Calculator {
    public static double num1, num2, result;
    public static void add(double num1, double num2){
        setNumbers(num1, num2);
        result =  num1 + num2;
    }
    public static void subtract(double num1, double num2){
        setNumbers(num1, num2);
        result = num1 - num2;
    }
    public static void divide(double num1, double num2) throws ArithmeticException{
        if(num2 == 0)
            throw new ArithmeticException("Can't divide " + num1 + " by zero!");
        setNumbers(num1, num2);
        result = num1 / num2;
    }
    public static void multiply(double num1, double num2){
        setNumbers(num1, num2);
        result = num1 * num2;
    }
    private static void setNumbers(double num1, double num2){
        Calculator.num1 = num1;
        Calculator.num2 = num2;
    }
}
